packages/babel-parser/src/parser/index.js

```diff
@@ -26,7 +26,7 @@ export default class Parser extends StatementParser {
     this.inModule = this.options.sourceType === "module";
     this.scope = new ScopeHandler(this.raise.bind(this), this.inModule);
     this.plugins = pluginsMap(this.options.plugins);
-    this.filename = options.sourceFilename;
+    this.filename = options.sourceFilename || options.sourceFileName;
```

Comment:
Since the nullish coaleshing proposal is now stage 3, we could enable `@babel/plugin-proposal-nullish-coalescing-operator` and use `options.sourceFilename ?? options.sourceFileName` :sunglasses:

Reply:
It has been enabled on master, you just need to rebase.